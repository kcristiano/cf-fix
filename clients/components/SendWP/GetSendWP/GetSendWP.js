import React from 'react';
import classNames from 'classnames';
import { __ } from "@wordpress/i18n";

/**
 * Create the UI for SendWP information and, link and Remote install
 * @param {Object} props
 * @return {*}
 * @constructor
 */
export const GetSendWP = (props) => {	


	return(
		<div className={classNames(props.className,GetSendWP.classNames.wrapper)} style={{margin:'20px'}}>
			<div className="caldera-forms-clippy-zone-inner-wrap" style={{background: 'white'}}>
				<div className="caldera-forms-clippy"
					style={{textAlign:'center',backgroundColor:'white',padding:'20px'}}>
					<h2>
						{__( 'Getting WordPress email into an inbox just got a lot easier!', 'caldera-forms' )}
					</h2>
					<p>
						{__(
							'SendWP makes getting emails delivered as simple as a few clicks. So you can relax, knowing those important emails are being delivered on time.',
							'caldera-forms'
						)}
					</p>
					<a href="https://sendwp.com?utm_source=Caldera+Forms+Plugin&utm_medium=Forms_Email+Settings&utm_campaign=SendWP+banner+ad"
						target="_blank" className="bt-btn btn btn-green" style={{width: '80%',margin: 'auto', paddingTop: '12px' }}>
						{__( 'Learn More', 'caldera-forms' )}
					</a>
				</div>
			</div>
		</div>
	)
};

/**
 * Class names used in the GlobalForms settings component
 * @type {{wrapper: string}}
 */
GetSendWP.classNames = {
	wrapper: 'caldera-forms-global-form-settings'
};