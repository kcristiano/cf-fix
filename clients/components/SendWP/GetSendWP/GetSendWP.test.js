import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme/build';
import Enzyme from 'enzyme/build';
import Adapter from 'enzyme-adapter-react-16/build';
import {GetSendWP} from "./GetSendWP";

Enzyme.configure({ adapter: new Adapter() });

describe( 'GetSendWP component', () => {
	it( 'Matches snapshot with minimal props', () => {
		expect(
			renderer.create(
				<GetSendWP/>
			).toJSON()
		).toMatchSnapshot()
	});

	it( 'Is wrapped in the right class', () => {
		expect(
			shallow(
				<GetSendWP/>
			).find( '.' + GetSendWP.classNames.wrapper )
				.length
		).toEqual(1)
	});
});