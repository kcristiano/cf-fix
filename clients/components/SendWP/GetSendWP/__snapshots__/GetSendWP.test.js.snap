// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetSendWP component Matches snapshot with minimal props 1`] = `
<div
  className="caldera-forms-global-form-settings"
  style={
    Object {
      "margin": "20px",
    }
  }
>
  <div
    className="caldera-forms-clippy-zone-inner-wrap"
    style={
      Object {
        "background": "white",
      }
    }
  >
    <div
      className="caldera-forms-clippy"
      style={
        Object {
          "backgroundColor": "white",
          "padding": "20px",
          "textAlign": "center",
        }
      }
    >
      <h2>
        Getting WordPress email into an inbox just got a lot easier!
      </h2>
      <p>
        SendWP makes getting emails delivered as simple as a few clicks. So you can relax, knowing those important emails are being delivered on time.
      </p>
      <a
        className="bt-btn btn btn-green"
        href="https://sendwp.com?utm_source=Caldera+Forms+Plugin&utm_medium=Forms_Email+Settings&utm_campaign=SendWP+banner+ad"
        style={
          Object {
            "margin": "auto",
            "paddingTop": "12px",
            "width": "80%",
          }
        }
        target="_blank"
      >
        Learn More
      </a>
    </div>
  </div>
</div>
`;
